execution_engine/src/core/engine_state/mod.rs

```diff
@@ -2376,6 +2377,48 @@ where
             .map_err(Error::from)
     }
 
+    pub fn read_trie(
+        &self,
+        correlation_id: CorrelationId,
+        trie_key: Blake2bHash,
+    ) -> Result<ReadTrieResult, Error>
+    where
+        Error: From<S::Error>,
+    {
+        let maybe_trie: Option<Trie<Key, StoredValue>> =
+            self.state.read_trie(correlation_id, &trie_key)?;
```

Comment:
@henrytill Since we don't have a state root here when we need to call `read_trie`, I ended up moving it to the `StateProvider` trait